lib/add-nested-value.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const addNestedValue = function (pojo, name, value) {
+  const recurse = function recurse(pojo, keys, value) {
```

Comment:
```js
const recurse = function recurse(pojo, name, value) {
...
};
```

The part after the equals sign in the above isn't a function declaration, it's a named function expression.  You have to use a name in the expression to call the function recursively because the variable's value is undefined until the semicolon is reached.  Named function expressions were added to replace `arguments.callee`.  See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee.

Is this what your were asking about?